### ISSUE TITLE
Setting `numpy=True` causes timezone to be dropped

### ISSUE THREAD

**Issue by kforeman** [snowflakedb/snowflake-sqlalchemy#84]
We have tried to switch from `snowflake-connector` to `snowflake-sqlalchemy` in order to take advantage of some of its features (numpy compatibility, connection pool management through engines, querying directly into Pandas using `pd.read_sql_query()`, but we're running into an issue that we can't both get numpy-compatible query results and timezones simultaneously.

Turning on `numpy=True` when creating the engine fixes our problem with decimals being objects instead of floats, but then we stop getting timezones.

[This notebook](https://gist.github.com/kforeman/b3291657ef762f94474252923f46665e) contains a reproducible example.

1. What version of Python are you using?
`Python 3.7.1`

2. What operating system and processor architecture are you using?
`Darwin-18.2.0-x86_64-i386-64bit`

3. What are the component versions in the environment? 
```
pandas
	0.23.4
snowflake.connector
	1.7.6
snowflake.sqlalchemy
	1.1.10
sqlalchemy
	1.2.15
```

4. What did you do?
I tried to find a combination of parameters to `snowflake.sqlalchemy.URL` that would both return numpy-friendly results (e.g. cast decimal to float) and attach timezones to the results.

5. What did you expect to see?
Setting `numpy=True` would return decimal cast to float, but still include timezones in timestamps.

6. What did you see instead?
Turning on `numpy=True` when creating the engine fixes our problem with decimals being objects instead of floats, but then we stop getting timezones.

**Comment by kforeman** [comment]
Looking back at it now, I see that using only the engine with numpy turned off will give us what we want. But for legacy code that expects a connection (stuff that was written previously and used `snowflake-connector`) we're using `engine.raw_connection()` which still has the issue.

**Comment by smtakeda** [comment]
That's a known limitation. When I implemented numpy datatype support, there was no timezone associated with `datetime64`. Will look into it...

**Comment by keller00** [comment]
@kforeman could you verify what you'd like to achieve?
I think fully supporting `datetime64` would be desirable, but that will not solve your backwards compatibility problem, it'd do the opposite.

**Comment by kforeman** [comment]
@keller00 the main concern is that using 
```python
engine = sqlalchemy.create_engine(snowflake.sqlalchemy.URL(
    account=account,
    database=database,
    user=user,
    authenticator=auth,
    schema=schema,
    numpy=True
))
pd.read_sql_query("<query with datetime columns with timezones>", engine)
```
strips the timezones from the results. It successfully converts decimal to float, but leaves us with broken datetimes. 

Using either the older `snowflake-connector` package or `sqlalchemy.create_engine(snowflake.sqlalchemy.URL(..., numpy=False)).raw_connection()` with the newer package will still give us timezones, but it leaves the decimals as objects instead of converting to float.

At this point we've got a workaround in place, so I'm less concerned with backwards compatibility.  

Thanks!

**Comment by keller00** [comment]
Okay, I'll work on that engine giving you timezones, as it should

**Comment by keller00** [comment]
Could you tell me the numpy version you have installed?

**Comment by kforeman** [comment]
```
pandas
	0.23.4
snowflake.connector
	1.7.6
snowflake.sqlalchemy
	1.1.10
sqlalchemy
	1.2.15
numpy
	1.15.4
```

**Comment by keller00** [comment]
As it says here: https://docs.scipy.org/doc/numpy/reference/arrays.datetime.html
> For backwards compatibility, datetime64 still parses timezone offsets, which it handles by converting to UTC. However, the resulting datetime is **timezone naive**

I'm not sure whether `numpy=True` should return a `numpy.datetime64` with a timezone attached to it. However; it seems like you can just assume UTC when you get a `numpy.datetime64` without a timezone.
This seems to be done by `pandas` and not `numpy` when you set `numpy=''`.

**Comment by keller00** [comment]
Newest release of `snowflake-connector-python` (1.7.10) should have fix for this issue, could you please verify @kforeman ?

**Comment by smtakeda** [comment]
This was fixed in 1.7.11
